app/services/most_recently_used_searcher.rb

```diff
@@ -0,0 +1,21 @@
+class MostRecentlyUsedSearcher
+  
+  attr_reader :user
+
+  def initialize(user)
+    @user = user
+  end
+
+  def recently_used_facilities(limit = 5)
+    return [] unless user
+
+    Facility.active.joins(products: { order_details: :order }).merge(Order.for_user(user).group("orders.facility_id").order('MAX(orders.ordered_at) DESC')).limit(limit)
```

Comment:
Is there a good way to add some line breaks in here to make the line not so long and easier to read?

Also, use single quotes. And can you say `group(orders: :facility_id)`? (It’s very likely that doesn’t work, but would probably be cleaner)

Reply:
So `group(orders: :facility_id)` doesn't work (the group method doesn't accept hash arguments) but `group(:id)` does the trick.

Reply:
I struggle with understanding why `group(:id)` works. Might just be how the two scopes get merged. I think I would lean towards keeping the string in that case.

Reply:
@meara it looks like Oracle doesn't like the new AR queries so we're gonna have to break it up into two steps like it was before.